### PR TITLE
Add option for extra containers

### PR DESCRIPTION
**Pull request by ianstanton**
## Description
- Support for adding extra containers to webserver, scheduler, worker pods

Example usage:
```
workers:
  extraContainers:
    - name: my-container
      image: foo/bar:tag
      imagePullPolicy: IfNotPresent
      args: 
      - "foo"
      - "bar"
scheduler:
  extraContainers:
    - name: my-container
      image: foo/bar:tag
      imagePullPolicy: IfNotPresent
      args: 
      - "foo"
      - "bar"
webserver:
  extraContainers:
    - name: my-container
      image: foo/bar:tag
      imagePullPolicy: IfNotPresent
      args: 
      - "foo"
      - "bar"
```

## 🎟 Issue(s)

Resolves https://github.com/astronomer/issues/issues/2683

## 🧪  Testing

> What are the main risks associated with this change, and how are they addressed by tests added in this PR?

> Please add helm unit testing, if applicable. The docs are regretfully sparse for helm unit testing, [here](https://github.com/astronomer/helm-unittest/blob/main/DOCUMENT.md) is what we have to work with. In general, these kind of tests can be used for specific assertions like 'when these values are set, the resource ends up looking like XYZ' but not for generalized assertions like 'for all resources, make sure the namespace is not set to the release name'.

> Please also add to the system testing [here](../tests/functional-tests). This kind of testing allows you to connect into any live, running pod to make assertions about how they are operating.

## 📋 Checklist

- [ ] The PR title is informative to the user experience
- [ ] Functional test(s) added
- [ ] Helm chart unit test(s)

### CODE REVIEW

**Review by kaxil — COMMENTED** [review]
looks good, can we add some tests too please?

**Comment by jedcunningham** [comment]
No REAME.md changes? Should we toss `workers.extraContainers` into `pod-template-file.yaml` too?

**Review by jedcunningham — COMMENTED** [review]
No REAME.md changes? Should we toss workers.extraContainers into pod-template-file.yaml too?

File: README.md [hunk 1]
```diff
@@ -164,6 +164,7 @@ The following tables lists the configurable parameters of the Airflow chart and
 | `workers.targetCPUUtilization`                        | Target CPU Utilization of workers                                                                            | `80`                                              |
 | `workers.targetMemoryUtilization`                     | Target Memory Utilization of workers                                                                         | `80`                                              |
 | `workers.safeToEvict`                                 | Allow Kubernetes to evict worker pods if needed (node downscaling)                                           | `true`                                            |
+| `workers.extraContainers`                             | Add additional containers to worker pod(s)                                                                   | `[]`                                              |
```
Comment (kaxil) [inline README.md:167]: Maybe add something about it applies to Pod template file too, here and in values.yaml

```suggestion
| `workers.extraContainers`                             | Add additional containers to worker pod(s) including `pod_template_file`                                                                 | `[]`                                              |
```

File: README.md [hunk 2]
```diff
@@ -188,6 +190,7 @@ The following tables lists the configurable parameters of the Airflow chart and
 | `webserver.jwtSigningCertificateSecretName`           | Name of secret to mount Airflow Webserver JWT singing certificate from                                       | `~`                                               |
 | `webserver.defaultUser`                               | Optional default airflow user information                                                                    | `{}`                                              |
 | `webserver.useDefaultAirflowImage`                    | Prevent Airflow webserver reboot only for Airflow 2.0.0                                                      | `false`                                           |
+| `webserver.extraContainers`                           | Add additional containers to webserver pod                                                                   | `[]`                                              |
```
Comment (jedcunningham) [inline README.md:193]: Nit: no plural here?

**Review by jedcunningham — APPROVED** [review]

**Review by kaxil — APPROVED** [review]
Looks good, though I think one test is missing to check if extra containers are added to `pod_template_file`

**Comment by ianstanton** [comment]
> Looks good, though I think one test is missing to check if extra containers are added to `pod_template_file`

Added

File: README.md (hunk 2 above)
Comment (ianstanton) [reply README.md]: Fixed

**Review by kaxil — APPROVED** [review]

**Pull request merged**